packages/core/lib/build.js

```diff
@@ -82,10 +76,8 @@ const Build = {
       builder = new CommandBuilder(builder);
     } else if (typeof builder !== "function") {
       if (builder.build == null) {
-        return callback(
-          new BuildError(
-            "Build configuration can no longer be specified as an object. Please see our documentation for an updated list of supported build configurations."
-          )
+        throw new BuildError(
+          "Build configuration can no longer be specified as an object. Please see our documentation for an updated list of supported build configurations."
         );
       }
```

Comment:
The logic for resolving `builder` actions looks it can be simplified.

```
    if (builder === null || typeof builder === "undefined") {
      logger.log(
        "No build configuration found. Preparing to compile contracts."
      );
    } else if (typeof builder === "string") {
      builder = new CommandBuilder(builder);
    } else if (typeof builder === "function") {
      builder = { build: builder };
    } else if (builder.build == null) {
        throw new BuildError(
          "Build configuration can no longer be specified as an object. Please see our documentation for an updated list of supported build configurations."
        );
    }
```